assessments/projects/knative/self-assessment.md

```diff
@@ -0,0 +1,577 @@
+
+### Actors
+**Kubernetes apiserver and ecosystem()**(https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/#:~:text=Synopsis,which%20all%20other%20components%20interact.):The API server in Kubernetes is responsible for verifying and setting up data for various API objects such as pods, services, replication controllers, and more. It handles RESTful operations and serves as the interface to the cluster's collective state, enabling interaction among all other components.
+
+(suggestions from the maintainers: Kubernetes apiserver and ecosystem (existing webhooks, RBAC, etc) -- this exists and forms the basis for managing the custom resources that drive Knative Serving.  Ability to do arbitrary writes here is basically a full compromise of Knative (and probably the Kubernetes cluster).
+
+1. Rest operations and example API options --advertise-address string:The IP address on which to advertise the apiserver to members of the cluster. This address must be reachable by the rest of the cluster. If blank, the --bind-address will be used. If --bind-address is unspecified, the host's default interface will be used.
+
+
+2. Kube Scheduler(https://kubernetes.io/docs/concepts/overview/components/#kube-scheduler):The control plane component monitors for new Pods that have not been allocated to a node and assigns a node for their execution.When deciding where to schedule these Pods, it considers various factors: the resource needs of both single and multiple Pods, constraints related to hardware, software, and policies, preferences for and against specific node affinities, the location of the required data, the potential for interference between different workloads, and any specific deadlines that need to be met.
+
+
+3. kube-controller-manager(https://kubernetes.io/docs/concepts/overview/components/#kube-scheduler):Control plane component that runs controller processes.Logically, each controller is a separate process, but to reduce complexity, they are all compiled into a single binary and run in a single process.
+
+
+**Serving Controller** (Service) (https://knative.dev/docs/serving/): This is a central component in Knative responsible for orchestrating the deployment and scaling of serverless workloads. It operates independently, managing the lifecycle of Knative services and their revisions. The Serving Controller is isolated in such a way that even if it encounters issues or vulnerabilities, these do not directly compromise the serverless workloads it manages.
+
+（suggestions from the maintainers: Knative Serving controllers and webhooks -- these manage most of the fields in the custom resources, including realizing lower-level Kubernetes resources (e.g. Deployments, k8s Services, certificate requests, etc).  These drive higher-level intents to lower-level implementations.  These implement all the public Knative Serving resources (Service, Route, Configuration, Revision)).
```

Comment:
Can you please clarify what does "suggestions from the maintainers" mean? 
What and Who are the suggestions for? 
Is it additional information maintainers would like to convey?

Reply:
Thank you for your feedbacks! Those are the suggestions we got from the maintainer Evan Anderson. I included here as a skeleton for the actors part as people might have different views on which part is qualified as actors. Let me know if you want me to take that off.